pkg/client/github_client.go

```diff
@@ -0,0 +1,26 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/v52/github"
+	"golang.org/x/oauth2"
+)
+
+// GitHubAPICallDelay it's used to "slow" down the number of requests we perform to GitHub API , in order to avoid rate limit issues.
+const GitHubAPICallDelay = 1 * time.Minute
```

Comment:
let's either move this delay to the controller's code, or move the logic here. having the delay and the logic at two places can be complicated.

Reply:
I've moved the logic here, since we will have to use the same function in member-operator, this way we avoid some duplication (even if it's just few lines of code). Plz see: https://github.com/codeready-toolchain/toolchain-common/pull/315/commits/8dc20d30637d503651bc415fed8c453b4c518a4e